compare-builds.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+# Tests to compare 2 SDK builds - generally a Microsoft-built SDK and
+# a source-build SDK. This should help us find issues/surprises before
+# we ship out a source-built SDK.
+#
+# See --help for more details, including usage
+
+set -euo pipefail
+IFS=$'\n\t'
+
+# filter out some directory that are special and source-build only:
+# - reference-packages/
+# - source-built-artifacts/
+paths_to_ignore_regex='\.\/reference-packages|\./source-built-artifacts'
```

Comment:
Huh... do these show up in e.g. `dotnet-sdk-3.1.104....tar.gz`? Or is this special for distro builds as the way they're being preserved for N+1 or something like that?

Reply:
These are special in our distro builds; it's where `source-build-reference-packages` gets installed to. See https://github.com/dotnet/source-build/issues/1456#issuecomment-578864169

Reply:
Ah, got it, didn't catch the diff you linked was between essentially two `DOTNET_ROOT`s (based on how it's named).

Reply:
The header should be even better now at making this more explicit. For example:

```
--- /home/omajid/downloads/dotnet-sdk-3.1.103-linux-x64.tar.gz
+++ /usr/lib64/dotnet
```